ChiTech/ChiPhysics/FieldFunction/lua/physics_getfieldfunclist.cc

```diff
@@ -55,3 +55,54 @@ int chiGetFieldFunctionList(lua_State* L)
 
   return 2;
 }
+
+
+//###################################################################
+/**Gets a field-function handle by name.
+\param FFname string Name of the field function.
+
+\return handle If the field-function was found and a handle identified the valid
+               handle will be returned (i.e., a natural number >= 0). If the
+               field-function by the given name was not found then the function
+               will return null.*/
+int chiGetFieldFunctionHandleByName(lua_State* L)
+{
+  const std::string fname = __FUNCTION__;
+  int num_args = lua_gettop(L);
+  if (num_args != 1)
+    LuaPostArgAmountError(fname, 1, num_args);
+
+  LuaCheckNilValue(fname, L, 1);
+  LuaCheckStringValue(fname, L ,1);
+
+  const std::string ff_name = lua_tostring(L,1);
+
+  size_t ff_handle_counter = 0;
+  std::vector<size_t> handles_that_matched;
+  for (auto& pff : chi_physics_handler.fieldfunc_stack)
+  {
+    if (pff->text_name == ff_name)
+      handles_that_matched.emplace_back(ff_handle_counter);
+    ++ff_handle_counter;
+  }
+
+  size_t num_handles = handles_that_matched.size();
+
+  if (num_handles == 0)
+  {
+    chi_log.Log(LOG_0WARNING) << fname << ": No field-functions were found that "
+                              << "matched the requested name. A null handle will "
+                              << "be returned.";
+    lua_pushnil(L);
+    return 1;
+  }
+
+  if (num_handles > 1)
+    chi_log.Log(LOG_0WARNING) << fname << ": A total of " << num_handles
+                              << " field-functions were found that matched the "
+                              << " requested name. Only the first match will be "
+                              << " returned.";
+
+  lua_pushinteger(L, static_cast<lua_Integer>(handles_that_matched.front()));
+  return 1;
+}
```

Comment:
As you mentioned, more thought will need to be given to field function naming in the future to prevent this type of behavior that could easily yield an undesired result. Perhaps a check for duplicate named field functions or prepended solver names to field function names would work provided no duplicate solver names are allowed. For now, this works just fine though.

Reply:
Yes. Prepending the solver-name is on my thoughts. Still thinking about but I think solver names should become mandatory. And we need to think how to handle duplicate names.

Reply:
My opinion: do not allow duplicate field names. Here, the name is used as a key; keys should be unique. If the key is not unique, then the concept of retrieving by key has no meaning, as additional information is required to identify fully the entry to which the key corresponds in the event of multiple occurrences of the non-unique key. Of course, it is possible to provide a mechanism to return either all occurrences or to return the n-th occurrence, but that assumes that the user knows how to interpret the ordering, which may not be case.